test/test_buffer.cpp

```diff
@@ -139,6 +139,11 @@ BOOST_AUTO_TEST_CASE(destructor_templated_callback)
         boost::compute::buffer buf(context, 128);
         buf.set_destructor_callback(boost::bind(destructor_templated_callback_function, &invoked));
     }
+
+    // NVIDIA calls the callback in a background thread and so the check is executed
+    // before the callback function is called
+    sleep(1);
```

Comment:
Yeah it's a dirty fix. But NVIDIA calls the callback from a different thread and the check is executed before the callback was hit. In this case the test fails, the normal destructor callback tests works without problems. I think that the pure c callback doesn't need so much "computation" until the callback is hit.
You're right that the arbitrary amount of time isn't a good indicator, do maybe have another idea? Because somehow you have to wait. As long as the callback is asynchronous, I don't know a better way to wait for this callback to get called.